### ISSUE TITLE
Incorrect calculation of days in Period format

### ISSUE THREAD

**Issue by gs4096** [Joshua-Ferguson/time_machine#71]
There is an issue in time_machine library where the days in Period format are being incorrectly calculated. This issue seems to occur when the starting date falls on a day that is not part of a 31-day month (i.e., February, April, June, September, or November).

test code : 
void calcDate(DateTime sd, DateTime ed) {
  final df = DateFormat('yyyy-MM-dd');

  LocalDateTime _sd = LocalDateTime.dateTime(sd);
  LocalDateTime _ed = LocalDateTime.dateTime(ed);

  final p = _ed.periodSince(_sd);

  print('${df.format(ed)} - ${df.format(sd)} : ${p.years}Y ${p.months}M ${p.days}D');
}

  calcDate(DateTime(2022, 1, 31), DateTime(2023, 4, 24));
  calcDate(DateTime(2022, 2, 1), DateTime(2023, 4, 24));

  calcDate(DateTime(2022, 2, 28), DateTime(2023, 4, 24));
  calcDate(DateTime(2022, 3, 1), DateTime(2023, 4, 24));

  calcDate(DateTime(2022, 3, 31), DateTime(2023, 4, 24));
  calcDate(DateTime(2022, 4, 1), DateTime(2023, 4, 24));

  calcDate(DateTime(2022, 4, 30), DateTime(2023, 4, 24));
  calcDate(DateTime(2022, 5, 1), DateTime(2023, 4, 24));

  calcDate(DateTime(2022, 5, 31), DateTime(2023, 4, 24));
  calcDate(DateTime(2022, 6, 1), DateTime(2023, 4, 24));

result :

2023-04-24 - 2022-01-31 : 1Y 2M 24D
2023-04-24 - 2022-02-01 : 1Y 2M 23D
2023-04-24 - 2022-02-28 : 1Y 1M 27D  <- incorrect
2023-04-24 - 2022-03-01 : 1Y 1M 23D
2023-04-24 - 2022-03-31 : 1Y 0M 24D
2023-04-24 - 2022-04-01 : 1Y 0M 23D
2023-04-24 - 2022-04-30 : 0Y 11M 25D <- incorrect 
2023-04-24 - 2022-05-01 : 0Y 11M 23D
2023-04-24 - 2022-05-31 : 0Y 10M 24D 
2023-04-24 - 2022-06-01 : 0Y 10M 23D

**Comment by tigloo** [comment]
The calculation is correct. If you take the first example you marked as incorrect, you can calculate the solution stepwise:

2022-02-28 + 1Y = 2023-02-28
2023-02-28 + 1M = 2023-03-28
2023-03-28 + 27D = 2023-04-24

The other one is the same:

2022-04-30 + 11M = 2023-03-30
2023-03-30 + 25D = 2024-04-24

**Comment by gs4096** [comment]
Thank you, tigloo, for your response.
It's been so long since I asked my question that I had difficulty confirming the intent of my original query. 😊

There is actually only one day difference between March 1, 2022, and February 28, 2023. However, when the calculation is performed, it appears as a four-day difference.

Both results are correct according to your calculation method, but I believe I inquired because it generally seems to be an implausible result.

**Comment by tigloo** [comment]
You are correct that the calculation may seem to give implausible results. Working with `Period` and `LocalDateTime` will try to approximate results based on calendar math, so "2023-01-31 + 2 months" and "2023-01-31 + 1 month + 1 month" may actually yield different results. Calculations in local date/time can be ambiguous and not reversible.

There's a good explanation by the original NodaTime author here: https://blog.nodatime.org/2010/11/joys-of-datetime-arithmetic.html

If you are looking to get mathematically plausible durations from date differences, you can force the arithmetic to use `PeriodUnits.Days` or better yet work with `Instant` and only convert to `LocalDateTime` for display. Calculations with `Instant` are not ambiguous.

In case you are still using this package, I recommend to check out https://pub.dev/packages/time_machine2. This package seems to have stopped receiving updates and I tried to address most issues in that fork.

**Comment by gs4096** [comment]
Thank you for your kind response.

In my app, this package is used very restrictively. 
Regarding the issue mentioned, it is being handled separately through calculations. Based on your advice, I will also consider migrating to the time_machine2 package. 

Thank you once again.
